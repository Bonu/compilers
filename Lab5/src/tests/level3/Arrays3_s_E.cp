class Main {

  int la[50];
  int ia1[50];
  int ia2[50];
  string fa1[50];
  string fa2[50];

  int i = la["2.0"];
  int j = ia[la[10]] * 3;
  string a = fa1[ia1[2] * (i*j+i%j*"2.1")];
  string b = ia[3] + fa[a];

  int k = la[a];
  int m = la[i+j+a];
  int n = la[fa1[2]];

  bool p = fa1[ia1[la[ia2[a]]]] != fa2[ia1[b]];
  bool q = a[0] != fa2[ia1[1]];

  void main(){
  }
}
