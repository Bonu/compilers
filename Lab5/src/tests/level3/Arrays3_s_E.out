line 9: invalid array index -- requires int or char type 
           but found ` "2.0" ' which has type string
line 10: unable to resolve name -- ia
line 11: invalid argument 2 ` "2.1" ' for arithmetic operator `*'
          -- expects a numeric argument but found string
line 12: unable to resolve name -- ia
line 12: unable to resolve name -- fa
line 12: invalid array index -- requires int or char type 
           but found ` a ' which has type string
line 14: invalid array index -- requires int or char type 
           but found ` a ' which has type string
line 15: invalid argument 2 ` a ' for arithmetic operator `+'
          -- expects a numeric argument but found string
line 16: invalid array index -- requires int or char type 
           but found ` fa1 [ 2 ] ' which has type string
line 18: invalid array index -- requires int or char type 
           but found ` a ' which has type string
line 18: invalid array index -- requires int or char type 
           but found ` b ' which has type string
line 19: ` a ' is not an array variable
12 errors were detected
